lib/reso_transport/query.rb

```diff
@@ -80,17 +80,28 @@ def results
       resource.parse(results)
     end
 
-    # Can only be accessed after results call
+    # Can only be accessed after results call or if it's being set
     def next_link
       @next_link
     end
 
+    # used for setting the next_link with trestle's replication strategy
+    def next_link=(link)
+      @next_link = link
+    end
+
     def response
-      resource.get(compile_params)
+      next_link_or_params = using_trestle_replication? ? next_link : compile_params
+
+      resource.get(next_link_or_params)
     rescue Faraday::ConnectionFailed
       raise NoResponse.new(resource.request, nil, resource)
     end
 
+    def using_trestle_replication?
+      compile_params[:replication] && next_link.present?
```

Comment:
looks like this will be false on the first request regardless of adding the `replication` param or not.  Is that accurate?

Reply:
yes, that's the goal. the first request will never be true, since next link is not returned yet

Reply:
uh, maybe naming can be better then